unittests/savanna_disaster_recovery.cpp

```diff
@@ -0,0 +1,270 @@
+#include "savanna_cluster.hpp"
+
+using namespace eosio::chain;
+using namespace eosio::testing;
+
+BOOST_AUTO_TEST_SUITE(savanna_disaster_recovery)
+
+// ---------------------------------------------------------------------------------------------------
+BOOST_FIXTURE_TEST_CASE(node_goes_down, savanna_cluster::cluster_t) try {
+   auto& A=_nodes[0]; auto& C=_nodes[2];
+
+   C.close();
+   BOOST_REQUIRE_EQUAL(A.lib_advances_by([&]() { A.produce_blocks(4);  }), 4); // lib still advances with 3 finalizers
+   C.open();
+   A.push_blocks_to(C);
+   BOOST_REQUIRE_EQUAL(A.lib_advances_by([&]() { A.produce_blocks(4);  }), 4); // all 4 finalizers should be back voting
```

Comment:
The comment can be improved. Here only checks LIB advancing; the next checks C is back to vote.

Reply:
Yes, that comment really applies to the next line, and the ` // let's make sure of that` implies that it is only verified on that line.